internal/supported_types/embed.go

```diff
@@ -0,0 +1,35 @@
+package supported_types
+
+import (
+	"embed"
+
+	"gopkg.in/yaml.v2"
+)
+
+var awsSupported = make(map[string]interface{})
+
+type supportedInstanceTypes struct {
+	Aws struct {
+		Ec2 []string `yaml:"ec2"`
+	} `yaml:"aws"`
+}
+
+var supported supportedInstanceTypes
```

Comment:
Can you move this into the init function? It does not need to be at package level. This will save some memory since at the end of the init function it can be cleaned up by gc.